### DISCUSSION TITLE
python/function_except/

### DISCUSSION THREAD

**Discussion by giscus** [ambosing/ambosing.github.io#11] [Blog-Comments]
# python/function_except/

try!! 파이썬 예외 try!!

https://ambosing.github.io/python/function_except/

**Comment by mingeonho1** [comment]
이렇게 되면 각각 대부분의 함수에 try catch문이 붙을 것이고 그렇게 되면 예외 처리가 너무 많아질 수도 있어 헷갈릴 수도 있고 코드가 길어질텐데 그럴 때 개선할 수 있는 방법이 있을까요?

**Reply by ambosing** [reply]
꼭 모든 함수에서 try와 catch를 쓰지는 않아도 됩니다! 
그리고 에러가 발생해야하지 않는 부분 위에서는 예를 들어 divide 코드 부분을 보시면 volume < 0이라는 조건에서만 raise를 시켜 에러 처리를 하도록 하는 것도 가능합니다.
그리고 모든 함수에서 예외처리를 할 필요 없이 여기서는 커스텀한 예외처리를 꼭 해줘야겠다라고 생각되는 부분 최상위 예외에서 Exception을 잡아주고 해도 괜찮습니다.
